datajoint/user_relations.py

```diff
@@ -2,59 +2,83 @@
 Hosts the table tiers, user relations should be derived from.
 """
 
-import abc
 from .base_relation import BaseRelation
 from .autopopulate import AutoPopulate
 from .utils import from_camel_case
 from . import DataJointError
 
 
-class Part(BaseRelation, metaclass=abc.ABCMeta):
+class classproperty:
```

Comment:
Yeah I have no idea why there is no `classproperty`... Anyway, I'm glad to reintroduce `classproperty` for our use. Given that it's not standard, I still think it'd make sense to keep its use only within DataJoint internal, and not directly at user interfaces.